lib/tasks/gettext.rake

```diff
@@ -20,15 +20,23 @@ begin
     end
   end
 
-  desc 'Extract plugin strings - called via rake plugin:gettext[plugin_name]'
-  task 'plugin:gettext', :engine do |t, args|
-    @domain = args[:engine]
-    @engine = "#{@domain.camelize}::Engine".constantize
-    @engine_root = @engine.root
+  desc 'Extract plugin strings - called via rake plugin:gettext[engine]'
+  task 'plugin:gettext', [:engine] => [:environment] do |t, args|
+    unless args[:engine]
+      puts "You must specify the name of the plugin (e.g. rake plugin:gettext['my_plugin'])"
+      exit 1
+    end
+    path_name = args[:engine].tr('-', '_')
+    @engine = "#{path_name.camelize}::Engine".constantize
```

Comment:
Perhaps out of scope, but any thoughts on introducing an `engine` method on `Foreman::Plugin`?

Reply:
What do you mean? There already is `Foreman::Plugin#engine`